src/service/storage-impl.js

```diff
@@ -231,6 +254,16 @@ export class Store {
     return item ? item['v'] : undefined;
   }
 
+  /**
+   * @param {string} name
+   * @return {number|undefined}
+   */
+  getTimestamp(name) {
```

Comment:
I find it a bit unnecessary to have an individual `getTimestamp()` method, because one would always want to get timestamp along with the value. 

What about making `get()` accept an optional argument
```
get(name, opt_duration){}
```

Or another option is to have `Store.get(name)` always return the entire object. And then handle it within `Storage.get()`.